scripts/cot/botania/flowers/functional/echinacenko.zs

```diff
@@ -0,0 +1,63 @@
+/*
+Echinacenko (Echinacea + kenko[health]) - provides nutritions to player
+*/
+#loader contenttweaker
+
+import crafttweaker.world.IBlockPos;
+import crafttweaker.world.IWorld;
+import crafttweaker.util.Math;
+import crafttweaker.player.IPlayer;
+import mods.randomtweaker.cote.SubTileEntityInGame;
+import mods.contenttweaker.VanillaFactory;
+import crafttweaker.data.IData;
+import mods.randomtweaker.cote.ISubTileEntity;
+import mods.randomtweaker.cote.BlockAdded;
+import mods.randomtweaker.cote.ISubTileEntityGenerating;
+import mods.randomtweaker.cote.ISubTileEntityFunctional;
+import mods.contenttweaker.Color;
+
+static manaCost as int = 100;
+static nutritionGain as float = 1.0f;
+
+var echinacenko as ISubTileEntityFunctional = VanillaFactory.createSubTileFunctional("echinacenko", 16711935);
+echinacenko.maxMana = 1000;
+echinacenko.range = 4;
+echinacenko.onUpdate = function(subtile, world, pos) {
+    if(world.isRemote()
+    || world.time%100!=13) return;
+
+    findPlayerAndNutrish(world, pos, subtile);
+    return;
+};
+echinacenko.register();
+
+function findPlayerAndNutrish(world as IWorld, pos as IBlockPos, subtile as SubTileEntityInGame) as void{
+    for player in world.getAllPlayers() {
+        if(isNull(player)
+        || Math.abs(player.x - pos.x - 0.5) > 4
+        || Math.abs(player.y - pos.y) > 4
+        || Math.abs(player.z - pos.z - 0.5) > 4 ) continue;
+        if(subtile.getMana() < manaCost) return;
+        val data = player.nbt.ForgeCaps.memberGet('nutrition:nutrition');
+        if(isPlayerMissingNutrient(data, player)) addNutritientToPlayer(data, player, subtile);
+    }
+}
+
+function isPlayerMissingNutrient(data as IData, player as IPlayer) as bool{
+    return (data.fruit as float < 99.9f||data.protein as float < 99.9f||data.grain as float < 99.9f||data.vegetable as float < 99.9f||data.dairy as float < 99.9f);
```

Comment:
it might be a stupid question, but will `data` be `null` here?

Reply:
It could be, but that is modified by nutrition mod almost every time player uses hunger (so most of ticks). 

Anyway I'll make safeguard anyway then.